spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/ribbon/EnvBasedClientConfig.java

```diff
@@ -0,0 +1,100 @@
+package org.springframework.cloud.netflix.ribbon;
+
+import static com.netflix.client.config.CommonClientConfigKey.DeploymentContextBasedVipAddresses;
+
+import java.util.concurrent.TimeUnit;
+
+import com.netflix.client.config.CommonClientConfigKey;
+import org.springframework.boot.bind.RelaxedPropertyResolver;
+import org.springframework.core.env.ConfigurableEnvironment;
+
+import com.netflix.client.config.DefaultClientConfigImpl;
+import com.netflix.client.config.IClientConfigKey;
+
+/**
+ * 
+ * {@link com.netflix.client.config.IClientConfig} implementation that depends on Spring
+ * environment for its values instead of using Archaius
+ * 
+ * @author Biju Kunjummen
+ */
+public class EnvBasedClientConfig extends DefaultClientConfigImpl {
+
+	private RelaxedPropertyResolver propertyResolver;
+
+	public static final IClientConfigKey<String> RETRYABLE_STATUS_CODES = new CommonClientConfigKey<String>("retryableStatusCodes") {};
+
+	public EnvBasedClientConfig(ConfigurableEnvironment environment) {
+		super();
+		this.propertyResolver = new RelaxedPropertyResolver(environment);
+	}
+	
+	@Override
+	public void loadProperties(String restClientName) {
+		setClientName(restClientName);
+		loadDefaultValues();
+	}
+
+	@Override
+	public void loadDefaultValues() {
+		super.loadDefaultValues();
+		putDefaultStringProperty(DeploymentContextBasedVipAddresses, this.getClientName());
+		putDefaultStringProperty(RETRYABLE_STATUS_CODES, "");
```

Comment:
The implementation is yet to be cleaned up @spencergibb :-), right now it extends `DefaultClientConfigImpl` and tries to let it do as much as possible - `loadDefaultValues` comes from `DefaultClientConfigImpl` and loads up all the default values for a ribbon specific client name - essentially it tries to fix up the precedence, anything with `clientname.ribbon.*` has a higher precedence than `ribbon.*` which has higher precedence than defaults built into the code.

If approach looks okay still, I can directly implement the interface `IClientConfig` instead and clean up this implementation.